subsys/ipc/ipc_service/lib/ipc_static_vrings.c

```diff
@@ -175,7 +142,20 @@ int ipc_static_vrings_init(struct ipc_static_vrings *vr, unsigned int role)
 	metal_io_init(vr->shm_device.regions, (void *) vr->shm_addr,
 		      vr->shm_physmap, vr->shm_size, -1, 0, NULL);
 
-	err = libmetal_setup(vr);
+	err = metal_register_generic_device(&vr->shm_device);
+	if (err != 0) {
+		return err;
+	}
+
+	err = metal_device_open("generic", vr->shm_device.name, &device);
+	if (err != 0) {
+		return err;
+	}
+
+	vr->shm_io = metal_device_io_region(device, 0);
+	if (vr->shm_io == NULL) {
+		return err;
```

Comment:
here `err == 0`  seems that issue was already present in  `libmetal_setup` and removed in next commit